devito/types/tensor.py

```diff
@@ -209,13 +209,14 @@ def div(self, shift=None, order=None, method='FD', **kwargs):
         comps = []
         func = vec_func(self)
         ndim = len(self.space_dimensions)
+        space_dims = [d.root for d in self.space_dimensions]
```

Comment:
the redundant changes in this file are a bit ugly, maybe we should change the semantics of `space_dimensions` to just return the root, or (prob better) add a new properaty